src/lorawan.cpp

```diff
@@ -131,52 +130,40 @@ void do_send(osjob_t* j){
     //mydata[5] = data  & 0xff;
 
     // Check if there is not a current TX/RX job running
+    u8x8.clearLine(7); 
     if (LMIC.opmode & OP_TXRXPEND) {
         ESP_LOGI(TAG, "OP_TXRXPEND, not sending");
-        u8x8.clearLine(7); 
         u8x8.drawString(0, 7, "LORA BUSY");
     } else {
         // Prepare upstream data transmission at the next possible time.
-        LMIC_setTxData2(1, mydata, sizeof(mydata)-1, (cfg.countermode & 0x02));
+        LMIC_setTxData2(1, mydata, sizeof(mydata), (cfg.countermode & 0x02));

```

Comment:
sure this works? i remember that there was some certain reason why i took sizeof(mydata)-1 here.
In the original sample code from IBM they also use -1.
Maybe it's because added termination '0' at end of string?

Reply:
yes you declared as a string of 4 char, so buffer was 4 char + 1 for \0 so 5 :-)